apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,221 @@
+            schema: 'signInUser(data: SignInUserInput!): SignInUserOutput',
+            doc: {
+                summary: 'This mutation authorizes the user by token, after confirming the phone number.\n\n' +
+                    'If the user is not registered, then he will be created with the data that is passed in the payload (user data).\n\n' +
+                    'If the existing user is missing some fields, then these fields will be taken from the payload (user data) and updated.\n\n' +
+                    'This mutation is not available for service users!',
+                errors: Object.fromEntries(Object.entries(ERRORS).filter(([, error]) => typeof error === 'object')),
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { confirmActionToken, userType, userData, dv, sender } = data
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                if (!SUPPORTED_USER_TYPES.includes(userType)) throw new GQLError(ERRORS.NOT_SUPPORTED_USER_TYPE, context)
+
+                if (!confirmActionToken) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                await checkDailyRequestLimitCountersByIp(context, 'signInUser', context.req.ip)
+
+                const action = await ConfirmPhoneAction.getOne(context,
+                    {
+                        token: confirmActionToken,
+                        expiresAt_gte: new Date().toISOString(),
+                        completedAt: null,
+                        isPhoneVerified: true,
+                        deletedAt: null,
+                    }
+                )
+                if (!action) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                const dvAndSender = { dv, sender }
+                const userInfo = {
+                    ...userData,
+                    type: userType,
+                    name: String(get(userData, 'name', '')).trim() || null,
+                    email: normalizeEmail(get(userData, 'email')) || null,
+                    isEmailVerified: false,
+                    phone: normalizePhone(get(userData, 'phone')) || null,
+                    isPhoneVerified: false,
+                }
+
+                if (!!userInfo.phone && userInfo.phone !== action.phone) throw new GQLError(ERRORS.DIFFERENT_PHONE_NUMBERS, context)
+                if (!action.phone) throw new GQLError(ERRORS.INVALID_PHONE_NUMBER, context)
+
+                userInfo.phone = action.phone
+                userInfo.isPhoneVerified = action.isPhoneVerified
+
+                let user = await getByCondition('User', {
+                    type: userInfo.type,
+                    phone: userInfo.phone,
+                })
+
+                if (user && user.deletedAt) throw new GQLError(ERRORS.USER_NOT_FOUND, context)
+
+                const userPayload = getUserPayload(user, userInfo)
```

Comment:
Confuse naming. It something like `prepareCreateOrUpdateUserData`

Reply:
And you can add dv, sender data here

Reply:
refactored